### PR TITLE
Add localdev documentation on accessing etcd (#3210)

### PR DESCRIPTION
**Pull request by richscott**
Add documentation on accessing the etcd instance in a localev setup (a Kind cluster). This shows the steps for deploying a pod (named "etcdclient") so that users can go into that pod and run `etcdctl` (or whatever etcd client they want), to interact directly with the etcd instance in a Kind cluster.

### CODE REVIEW

File: docs/developer/manual-localdev.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Here, we give an overview of a development setup for Armada that gives users ful
 
 Before starting, please ensure you have installed [Go](https://go.dev/doc/install) (version 1.20 or later), gcc (for Windows, see, e.g., [tdm-gcc](https://jmeubank.github.io/tdm-gcc/)), [mage](https://magefile.org/), [docker](https://docs.docker.com/get-docker/), [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl), and, if you need to compile `.proto` files, [protoc](https://github.com/protocolbuffers/protobuf/releases).
 
-For a full lust of mage commands, run `mage -l`.
+For a full list of mage commands, run `mage -l`.
```
Comment (suprjinx) [inline docs/developer/manual-localdev.md:7]: 😆

File: internal/armada/server/submit_to_log.go [hunk 2]
```diff
@@ -261,6 +261,12 @@ func (srv *PulsarSubmitServer) SubmitJobs(grpcCtx context.Context, req *api.JobS
 func (srv *PulsarSubmitServer) CancelJobs(grpcCtx context.Context, req *api.JobCancelRequest) (*api.CancellationResult, error) {
 	ctx := armadacontext.FromGrpcCtx(grpcCtx)
 
+	if req.JobSetId == "" || req.Queue == "" {
```
Comment (suprjinx) [inline internal/armada/server/submit_to_log.go:264]: this hunk (and others below) looks like it belongs to a different PR...?

**Comment by richscott** [comment]
Closing this PR, due to a mangled rebase adding many unrelated files to this PR; a new, clean PR will be created.
